src/components/CallbackSamples/CallbackSamples.tsx

```diff
@@ -0,0 +1,84 @@
+import { observer } from 'mobx-react';
+import * as React from 'react';
+
+import styled from '../../styled-components';
+import { RightPanelHeader } from '../../common-elements';
+import { RedocNormalizedOptions } from '../../services';
+import { CallbackModel } from '../../services/models';
+import { OptionsContext } from '../OptionsProvider';
+import { GenericDropdown } from '../GenericDropdown/GenericDropdown';
+import { DropdownOrLabel } from '../DropdownOrLabel/DropdownOrLabel';
+import { InvertedSimpleDropdown, MimeLabel } from '../PayloadSamples/styled.elements';
+import { CallbackReqSamples } from './CallbackReqSamples';
+
+export interface CallbackSamplesProps {
+  callbacks: CallbackModel[];
+}
+
+@observer
+export class CallbackSamples extends React.Component<CallbackSamplesProps> {
+  static contextType = OptionsContext;
+  context: RedocNormalizedOptions;
+
+  private renderDropdown = props => {
+    return <DropdownOrLabel Label={MimeLabel} Dropdown={InvertedSimpleDropdown} {...props} />;
+  };
+
+  render() {
+    const { callbacks } = this.props;
+
+    if (!callbacks || callbacks.length === 0) {
+      return null;
+    }
+
+    const operations = callbacks
+      .map(callback => callback.operations.map(operation => operation))
+      .reduce((a, b) => a.concat(b), []);
+
+    // Sums number of code samples per operation per callback
+    const numSamples = operations.reduce(
+      (sampleSum, operation) => sampleSum + operation.codeSamples.length,
+      0,
+    );
+
+    const hasSamples = numSamples > 0;
```

Comment:
Remove the reduce above and just use simple `Array.some` check

```suggestion
    const hasSamples = operations.some(operation => operation.codeSamples.length > 0);
```

Reply:
+ return early here (`if (!hasSamples) { return null }`)